pyanaconda/ui/tui/spokes/storage.py

```diff
@@ -426,7 +426,11 @@ def __init__(self, data, storage, payload, storage_module, partitioning):
         # Choose the initialization mode.
         self._disk_init_proxy = STORAGE.get_proxy(DISK_INITIALIZATION)
         self._orig_init_mode = self._disk_init_proxy.InitializationMode
-        self._init_mode = self._disk_init_proxy.InitializationMode
+        # Preserve clearpart mode from kickstart.
+        if self._orig_init_mode != CLEAR_PARTITIONS_DEFAULT:
+            self._init_mode = self._orig_init_mode
+        else:
+            self._init_mode = CLEAR_PARTITIONS_ALL
         self._init_mode_list = sorted(INIT_MODES.keys())
```

Comment:
I would reorganize the code a little to make it easier to follow. Also, the comment about the kickstart is not really relevant, because the property could be set directly via DBus.

```suggestion
        self._init_mode = self._orig_init_mode
        self._init_mode_list = sorted(INIT_MODES.keys())

        if self._init_mode == CLEAR_PARTITIONS_DEFAULT:
            self._init_mode = CLEAR_PARTITIONS_ALL
```